content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,315 @@
+
+- A client creates a `Socket` object, giving the server name and the connecting port number as the server awaits.
+
+- The `Socket` class constructor will attempt to connect the client to the server and port number provided. The client now has a socket object that can connect to the server if communication occurs.
+
+- The server's `accept()` method returns a reference to a new server socket connected to the client's socket.
+ 
+If the connection is successful, the client and server can then communicate using **I/O streams**. The client and server socket classes provide the **I/O streams**. The client's OutputStream communicates with the server's InputStream, and the server's OutputStream communicates with the client's InputStream.
+A stream is a collection of sequenced data, in case you didn't know. 
+
+#### There are two types of streams:
+1. A character stream (usually used with text files).
+2. A byte stream (usually used with images).
+
+A character stream is in human-readable language while a byte stream is in machine language, a language understandable by the CPU.
+
+>In this situation, the client and server will simultaneously broadcast messages to each other's streams. Because TCP is a two-way communication protocol, this is conceivable.
+
+#### Programming on the Server-side
+
+A `server` socket is a subclass of the ServerSocket that uses the constructor below:
+
+```java
+serversocket = new ServerSocket(int port)
+```
+Port: the port number on which the server will listen for the client's requests.
+
+>The ServerSocket object stays idle, waiting for network requests. Our client will attempt to connect to port 5000, which we have specified.
+
+**Note** :
+Make sure the ports are the same otherwise, the connection will fail.
+
+There will be two while loops in use:
+
+**First Loop** - This ensures that the server is indeed running.
+**Second Loop** - Ensures that the server interacts with the client after connection until the client disconnects.
+
+```java
+while (true)
+{  
+    try {  
+   socket = serversocket.accept();  
+   
+   inputStreamReader = new InputStreamReader(socket.getInputStream());  
+   outputStreamWriter = new OutputStreamWriter(socket.getOutputStream()); 
+   
+   bufferedReader = new BufferedReader(inputStreamReader);  
+   bufferedWriter = new BufferedWriter(outputStreamWriter);  
+  while (true) { 
+  String msgFromClient = bufferedReader.readLine();  
+  
+  System.out.println("Client: " + msgFromClient); 
+  bufferedWriter.write(" MSG Received");  
+  bufferedWriter.newLine();  
+  bufferedWriter.flush();  
+  
+ if (msgFromClient.equalsIgnoreCase("BYE"))  
+                break;  
+  
+  }
+      socket.close();  
+      inputStreamReader.close();  
+      outputStreamWriter.close();  
+      bufferedReader.close();  
+      bufferedWriter.close();  
+  
+    } catch (IOException e) {  
+        e.printStackTrace();  
+  }  
+}
+```
+   
+#### Methods used in ServerSocket
```

Comment:
There are no `getInputStream` and `getOutputStream` methods in the `ServerSocket` class but this header seems to imply that. Maybe a more appropriate header would be:

> \#\#\#\# Methods used in the server-side code